R/simplify_network.R

```diff
@@ -88,11 +85,11 @@ simplify_network = function(rnet_y, parameters, region_boundary) {
   rnet_merged_all_projected_buffer = sf::st_as_sf(rnet_merged_all_geos_buffer)
 
   # Transform the coordinate reference system of 'rnet_merged_all' to WGS 84 (EPSG:4326).
-  rnet_merged_all_buffer = sf::st_transform(rnet_merged_all_projected_buffer, "EPSG:4326")
+  # rnet_merged_all_buffer = sf::st_transform(rnet_merged_all_projected_buffer, "EPSG:4326")
 
   # Subsetting another dataset 'rnet_y' based on the spatial relation with 'rnet_merged_all_buffer'.
   # It selects features from 'rnet_y' that are within the boundaries of 'rnet_merged_all_buffer'.
-  rnet_y_subset = rnet_y[rnet_merged_all_buffer, , op = sf::st_within]
+  rnet_y_subset = rnet_y[rnet_merged_all_projected_buffer, , op = sf::st_within]
```

Comment:
This line of code take 95% of time

Reply:
Make sense.

Reply:
2 ideas for making this faster:

1. Run the function on LA-sized (not region-sized) inputs, that will be done in #550 
2. Test to see if the equivalent code with `{geos}` functions is faster

Reply:
should be 

```suggestion
  rnet_y_subset = rnet_yp[rnet_merged_all_projected_buffer, , op = sf::st_within]
```

Right?